src/bootstrap/install.rs

```diff
@@ -221,27 +215,14 @@ install!((self, builder, _config),
     };
     RustAnalyzer, "rust-analyzer", Self::should_build(_config), only_hosts: true, {
         builder.ensure(dist::RustAnalyzer { compiler: self.compiler, target: self.target });
-        if Self::should_install(builder) {
-            install_rust_analyzer(builder, self.compiler.stage, self.target);
-        } else {
-            builder.info(
-                &format!("skipping Install rust-analyzer stage{} ({})", self.compiler.stage, self.target),
-            );
-        }
+        install_rust_analyzer(builder, self.compiler.stage, self.target);
```

Comment:
AFAIK rust-analyzer is not guaranteed to always build successfully. Therefore install should be performed only when build succeeds.

Reply:
rust-analyzer builds on stable today, so is indeed expected to always build successfully.